packages/sdk/src/messageBackend/goldsky.ts

```diff
@@ -0,0 +1,374 @@
+            m = await this.fetchMessage({
+                messageHash
+            });
+            if (m) {
+                this.storeMessage(m);
+            }
+        }
+
+        return m;
+    }
+
+    /**
+     * Get the message representation associated with this message (if any)
+     * by dispatch transaction
+     *
+     * @returns A message representation (if any)
+     */
+    async getMessageByTx(tx: string): Promise<GoldSkyMessage | undefined> {
+        let m: GoldSkyMessage | undefined;
+        let messageHash = this.dispatchTxToMessageHash.get(tx);
+        if (!messageHash) {
+            m = await this.fetchMessage({
+                transactionHash: tx
+            });
+            if (m) {
+                this.storeMessage(m);
+            }
+        }
+
+        return m;
+    }
+
+
+    /**
+     * Get the `Dispatch` transaction hash associated with this message (if any)
+     *
+     * @returns A dispatch tx (if any)
+     */
+    async dispatchTx(messageHash: string): Promise<string | undefined> {
+        let m = await this.getMessage(messageHash);
+        return m?.dispatch_tx;
+    }
+
+    /**
+     * Get the `Relay` transaction hash associated with this message (if any)
+     *
+     * @returns A relay tx (if any)
+     */
+    async relayTx(messageHash: string): Promise<string | undefined> {
+        let m = await this.getMessage(messageHash);
+        return m?.relay_tx;
+    }
+
+    /**
+     * Get the `Update` transaction hash associated with this message (if any)
+     *
+     * @returns A update tx (if any)
+     */
+    async updateTx(messageHash: string): Promise<string | undefined> {
+        let m = await this.getMessage(messageHash);
+        return m?.update_tx;
+    }
+
+    /**
+     * Get the `Process` transaction hash associated with this message (if any)
+     *
+     * @returns A relay tx (if any)
+     */
+    async processTx(messageHash: string): Promise<string | undefined> {
+        let m = await this.getMessage(messageHash);
+        return m?.process_tx;
+    }
+
+    /**
+     * Get message hash associated with this message (if any)
+     *
+     * @returns A message hash (if any)
+     */
+    async getMessageHash(tx: string): Promise<string | undefined> {
+        const m = await this.getMessageByTx(tx);
+        return m?.message_hash;
+    }
+
+    /**
+     * Calculates an expected confirmation timestamp from relayed event
+     *
+     * @returns Timestamp (if any)
+     */
+    async confirmAt(messageHash: string): Promise<Date | undefined> {
+        const m = await this.getMessage(messageHash);
+        if (m && m.relay_tx) {
+            let intTs;
+            try {
+                intTs = parseInt(m.relayed_at!); // may throw
+            } catch(e) {
+                return undefined
+            }
+            if (intTs <= 946684800000) {
+                throw new Error(`That could not be`);
+            }
+            const optimisticSeconds = 30*60*1000;
```

Comment:
We can get optimisticSeconds from the config per network

Reply:
Also, it's better to get this from the replica contract and then cache the value.  Only check if the transaction has been relayed or else it won't exist yet

Reply:
Addressed in the new commit